lib/srv/ctx.go

```diff
@@ -676,22 +678,87 @@ func (c *ServerContext) String() string {
 	return fmt.Sprintf("ServerContext(%v->%v, user=%v, id=%v)", c.ServerConn.RemoteAddr(), c.ServerConn.LocalAddr(), c.ServerConn.User(), c.id)
 }
 
-// ExecCommand takes a *ServerContext and extracts the parts needed to create
-// an *execCommand which can be re-sent to Teleport.
-func (c *ServerContext) ExecCommand() (*ExecCommand, error) {
-	var pamEnabled bool
-	var pamServiceName string
-	var pamUseAuth bool
+func (c *ServerContext) getPAMConfig() (*PAMConfig, error) {
+	if c.srv.Component() != teleport.ComponentNode {
+		return nil, nil
+	}
```

Comment:
I would return an error here, seems like a legitimate error if a non-node component tries to get PAM configuration.

Reply:
This code is apparently called by a non-node component somewhere because a similar check was in the old code and integration tests fail without it.

Reply:
Interesting, it should not be. Can you share the error?